src/main/java/com/minecolonies/coremod/network/messages/PermissionsMessage.java

```diff
@@ -239,6 +239,70 @@ public void messageOnServerThread(final AddPlayer message, final EntityPlayerMP
         }
     }
 
+    /**
+     * Message class for adding a player or fakePlayer to a permission set.
+     */
+    public static class AddPlayerOrFakePlayer extends AbstractMessage<AddPlayerOrFakePlayer, IMessage>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 44 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)